### DISCUSSION TITLE
Manually entered variables disappear in web UI

### DISCUSSION THREAD

**Discussion by aleeops** [apache/airflow#33033] [General]
### Apache Airflow version

2.6.3

### What happened

When I manually add variables via the web GUI (Admin > Variables), the variables disappear moments later. Could be anywhere from a couple of seconds to a couple of minutes. It takes a refresh of the page to notice the variables are gone. Airflow is running on Kubernetes via Helm but the container is NOT restarting and the manually added variables still disappear. The logs don't appear to show anything unusual.

This was not an issue about a week ago when we were on `2.5.3`. Rolling back to `2.5.3` worked but as I looked more into it, it seemed like the issue was directly related to a Dependabot change which upped the `apache-airflow-providers-postgres` version from `5.5.0` to `5.5.2`. https://github.com/apache/airflow/blob/main/airflow/providers/postgres/CHANGELOG.rst#552

I redeployed `2.6.3` with the `apache-airflow-providers-postgres` version rolled back to `5.5.0` and the issue has returned.

### What you think should happen instead

Manually entered variables should persist until the container is restarted. As it is now, the variables disappear on their own in a few seconds. 

### How to reproduce

Navigate to the variables list (Admin > Variables), enter at least one variable, wait a bit, refresh the page, and it will be gone. 

### Operating System

Debian GNU/Linux 11 (bullseye)

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==8.3.1
apache-airflow-providers-celery==3.2.1
apache-airflow-providers-cncf-kubernetes==7.3.0
apache-airflow-providers-common-sql==1.6.0
apache-airflow-providers-docker==3.7.1
apache-airflow-providers-elasticsearch==4.5.1
apache-airflow-providers-ftp==3.4.2
apache-airflow-providers-google==10.2.0
apache-airflow-providers-grpc==3.2.1
apache-airflow-providers-hashicorp==3.4.1
apache-airflow-providers-http==4.5.0
apache-airflow-providers-imap==3.2.2
apache-airflow-providers-mysql==5.1.1
apache-airflow-providers-odbc==4.0.0
apache-airflow-providers-postgres==5.5.0
apache-airflow-providers-redis==3.2.1
apache-airflow-providers-sendgrid==3.2.1
apache-airflow-providers-sftp==4.3.1
apache-airflow-providers-slack==7.3.1
apache-airflow-providers-sqlite==3.4.2
apache-airflow-providers-ssh==3.7.1

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by pankajkoti** [comment]
hi @alee68 

I got a bit confused with the description. Were you able to narrow down the issue to a specific version of Airflow or a specific version of Postgres provider?

Nevertheless, I could not reproduce this issue with Airflow 2.6.3 together with both the versions 5.5.0 and 5.5.2 of the apache-airflow-providers-postgres package. I was able to see that the variables are getting recorded in the postgres database immediately.

Can you provide the exact version of the provider and Airflow you feel the issue is present on? Also, on which versions of Airflow and the provider do you mean to say the issue does not happen?

Sidenote: Can you check if some settings have changed on your postgres instance? Like autocommit disabled or something like this?

**Comment by potiuk** [comment]
We need definitely more information. For sure postgres provider has nothing to do it, this must have been some coincidence, because the provider is not even used in case of Variable writing. More evidences are needed to help you - I converted it to a discussion as this seems to be troubleshooting of your problem.

My guess is (you forgot to mention it) what is your database configuration. My best guess is - the problem is that you are using the embedded Postgres (which is development only) and that you have too little resources (for example memory) which causes the postgres container to restart (You wrote about container  - which I presume it was about webserver). But very likely your Postgres container IS restarting. 

As explained in the docs, the embedded Postgres is only suitable for development and you should use some externally managed DB. The postgres we embed lacks all the features (including persistency). 

Please take a look at investigate that part and switched to a "real" database managed externally first (if you do not use it yet) and if it does not work, please gather some more evidence - check if the variables table gets populated for example. Also you might have some scripts or dags that might simply delet those variables, so it would be great if you check if you do not have any of those configured.

All that is based on guesses of course.
